azure-pipelines.yml

```diff
@@ -193,8 +194,11 @@ stages:
             export PYTHONHASHSEED=$(python -S -c "import random; print(random.randint(1, 4294967295))")
             echo "PYTHONHASHSEED=$PYTHONHASHSEED"
             stestr run
+            python ./verify_parallel_map.py
             popd
           displayName: 'Run tests'
+          env:
+            QISKIT_PARALLEL: FALSE
```

Comment:
It was necessary when I wrote this change, see the commit message on: https://github.com/Qiskit/qiskit-terra/pull/7658/commits/93ded7f38b80c50105d16c599848ab6b1f83c0fd for the details. But, I updated the logic here in https://github.com/Qiskit/qiskit-terra/pull/7658/commits/3d1d56172da13fec26ae355d7eb44bb22f6ea1f6 to not run multithreaded code when we're in `parallel_map()` so I think we might be able to get away without adding this. That being said doing this made CI noticeably faster because we're not running 2x processes everywhere anymore.

Reply:
This is good enough for me - my only minor concern is that we might fail to test some code that _should_ run in parallel now, but I think we should be able to override that anyway if necessary within the test suite.

Reply:
Yeah, we can just copy the model I used in `verify_parallel_map` to override the env variable, or alternatively we can always explicitly add on to that script too for things we want to ensure are run in parallel.